doc/crypto/SCT_new.pod

```diff
@@ -0,0 +1,163 @@
+
+Only SCT_VERSION_V1 is currently supported.
+
+=item * B<SCT_set_log_entry_type>() to set the type of certificate the SCT was issued for:
+
+B<CT_LOG_ENTRY_TYPE_X509> for a normal certificate.
+B<CT_LOG_ENTRY_TYPE_PRECERT> for a pre-certificate.
+
+=item * B<SCT_set0_log_id>() or B<SCT_set1_log_id>() to set the LogID of the CT log that the SCT came from.
+
+The former takes ownership, whereas the latter makes a copy.
+See RFC 6962, Section 3.2 for the definition of LogID.
+
+=item * B<SCT_set_timestamp>() to set the time the SCT was issued (epoch time in milliseconds).
+
+=item * B<SCT_set_signature_nid>() to set the NID of the signature.
+
+=item * B<SCT_set0_signature>() or B<SCT_set1_signature>() to set the signature itself.
+
+The former takes ownership, whereas the latter makes a copy.
+
+=item * B<SCT_set0_extensions>() or B<SCT_set1_extensions> to provide SCT extensions.
+
+The former takes ownership, whereas the latter makes a copy.
+
+=back
+
+Alternatively, the SCT can be pre-populated from the following data using
+B<SCT_new_from_base64>():
+
+=over
+
+=item * The SCT version (only SCT_VERSION_V1 is currently supported).
+
+=item * The LogID (see RFC 6962, Section 3.2), base64 encoded.
+
+=item * The type of certificate the SCT was issued for:
+
+B<CT_LOG_ENTRY_TYPE_X509> for a normal certificate.
+B<CT_LOG_ENTRY_TYPE_PRECERT> for a pre-certificate.
+
+=item * The time that the SCT was issued (epoch time in milliseconds).
+
+=item * The SCT extensions, base64 encoded.
+
+=item * The SCT signature, base64 encoded.
+
+=back
+
+B<SCT_set_source>() can be used to record where the SCT was found
+(TLS extension, X.509 certificate extension or OCSP response). This is not
+required for verifying the SCT.
+
+=head1 NOTES
+
+Some of the setters return int, instead of void. These will all return 1 on
+success, 0 on failure. They will not make changes on failure.
+
+Most of the setters will reset the validation status of the SCT to
+SCT_VALIDATION_STATUS_NOT_SET (see L<SCT_verify(3)>).
+
+B<SCT_set_source>() will call B<SCT_set_log_entry_type>() if the type of
+certificate the SCT was issued for can be inferred from where the SCT was found.
+For example, an SCT found in an X.509 extension must have been issued for a pre-
+certificate.
+
+B<SCT_set_source>() will not refuse unknown values.
+
+=head1 RETURN VALUES
+
+B<SCT_set_version>() returns 1 if the specified version is supported, 0 otherwise.
+
+B<SCT_set_log_entry_type>() returns 1 if the specified log entry type is supported, 0 otherwise.
+
+B<SCT_set0_log_id>() and B<SCT_set1_log_id> return 1 if the specified LogID is a
+valid SHA-256 hash, 0 otherwise. Aditionally, B<SCT_set1_log_id> returns 0 if
+malloc fails.
+
+B<SCT_set_signature_nid> returns 1 if the specified NID is supported, 0 otherwise.
+
+B<SCT_set1_extensions> and B<SCT_set1_signature> return 1 if the supplied buffer
+is copied successfully, 0 otherwise (i.e. if malloc fails).
+
+B<SCT_set_source> will always return 1.
```

Comment:
If we're going to give it a return value, then we should probably document it as returning 0 on failure (without specifying what kinds of failures). This way if we ever want it to fail in the future then even old code should be checking the return value. Alternatively if we are confident this will never fail, even in the future, then we could just make it void.